### PR TITLE
Internal API to get original data from homogen_table

### PR DESCRIPTION
**Pull request by michael-smirnov**
Created function ``detail::get_original_data()`` that allows getting an array with data located directly in the homogen table without any copies. This is required for Python API to get a mutable pointer from a resulting table without unnecessary data copies.

### CODE REVIEW

File: cpp/oneapi/dal/table/detail/homogen_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#include "oneapi/dal/table/backend/homogen_table_impl.hpp"
+
+namespace oneapi::dal::detail::v1 {
+
+array<byte_t> get_original_data(const homogen_table& t) {
+    using impl_t = backend::homogen_table_impl;
+    using wrapper_t = homogen_table_impl_wrapper<impl_t>;
+
+    const auto& pimpl = pimpl_accessor{}.get_pimpl(t);
+    auto* impl_raw_ptr = dynamic_cast<wrapper_t*>(pimpl.get());
```
[the remark this answers is not in the document]
Comment (michael-smirnov) [reply cpp/oneapi/dal/table/detail/homogen_utils.cpp]: Yes

**Review by PetrovKP — APPROVED** [review]

**Review by rlnx — APPROVED** [review]

**Comment by michael-smirnov** [comment]
Review: http://intel-ci.intel.com/eb98421c-6de6-f149-96e6-8cdcd4b723a1

**Pull request merged**